pkg/metrics/metrics.go

```diff
@@ -159,6 +159,108 @@ var (
 		Name:      "policy_l7_received_total",
 		Help:      "Number of total L7 received requests/responses",
 	})
+
+	// Status
+
+	// ControllersFailing is a count of failing controllers
+	ControllersFailing = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Name:      "controllers_failing_total",
+		Help:      "Number of faling controllers",
+	})
+
+	// IPv4Addresses is a count of allocated IPv4 addresses in the cluster
+	IPv4Addresses = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Name:      "ipv4_addresses_total",
+		Help:      "Number of allocated IPv4 addresses",
+	})
+
+	// IPv6Addresses is a count of allocated IPv6 addresses in the cluster
+	IPv6Addresses = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Name:      "ipv6_addresses_total",
+		Help:      "Number of allocated IPv6 addresses",
+	})
+
+	// UnreachableNodes is a count of nodes which have no ICMP or HTTP L3
+	// connectivity
+	UnreachableNodes = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Name:      "unreachable_nodes_total",
+		Help:      "Number of nodes that cannot be reached",
+	})
+
+	// UnreachableHealthEndpoints is a count of node endpoints which have no
+	// ICMP or HTTP L3 connectivity
+	UnreachableHealthEndpoints = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Name:      "unreachable_health_endpoints_total",
+		Help:      "Number of health endpoints",
+	})
+
+	// AvgNodesICMPLatency is an average ICMP latency to nodes in the
+	// cluster
+	AvgNodesICMPLatency = prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
For latency, you almost certainly want a histogram or summary: https://prometheus.io/docs/practices/histograms/.  I'm unfortunately not the best person to ask about complex metric types, but folks on the Prometheus IRC/mailing lists are usually super helpful.

In addition, calculating things like averages and maximums is best left to PromQL queries.  You'll be able to infer that information from a summary or histogram later.

Reply:
Yes, this looks like an incorrect understanding of how to use the Prometheus library.